### PR TITLE
[20091] Add macOS Github CI

### PR DESCRIPTION
**Pull request by EduPonz**
## Description

This PR adds:
- A reusable workflow for macOS CI
- A workflow for macOS CI on PR and push events
- A nightly workflow for macOS CI

It depends on the following PRs:

* eprosima/eprosima-ci#44

@Mergifyio backport 2.12.x 2.11.x 2.10.x 2.6.x

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- *N/A*: The code follows the style guidelines of this project. 
- *N/A*: Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- *N/A*: Any new/modified methods have been properly documented using Doxygen. 
- *N/A*: Changes are ABI compatible. 
- *N/A*: Changes are API compatible. 
- *N/A*: New feature has been added to the `versions.md` file (if applicable).
- *N/A*: New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A*: Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: .github/workflows/reusable-windows-ci.yml
Comment (JesusPoderoso) [inline .github/workflows/reusable-windows-ci.yml:164]: It would be nice if we add some comment here explicitly saying why those flags have been included here "manually" for some reason

Comment (JesusPoderoso) [reply .github/workflows/reusable-windows-ci.yml:164]:
```suggestion
          # Apart from default .metas cmake args, these args are needed for the Windows workflows
          cmake_args_default: ${{ inputs.cmake-args }} -DTHIRDPARTY_Asio=FORCE -DTHIRDPARTY_TinyXML2=FORCE -DTHIRDPARTY_fastcdr=OFF -DTHIRDPARTY_UPDATE=ON -DEPROSIMA_EXTRA_CMAKE_CXX_FLAGS="/MP /WX"
```

File: .github/workflows/reusable-mac-ci.yml
Comment (JesusPoderoso) [inline .github/workflows/reusable-mac-ci.yml:94]: Shall we address this before merging?

File: .github/workflows/reusable-mac-ci.yml
Comment (JesusPoderoso) [inline .github/workflows/reusable-mac-ci.yml:99]: I think we should remove this and address it in other PR

File: .github/workflows/reusable-mac-ci.yml
Comment (EduPonz) [reply .github/workflows/reusable-mac-ci.yml:94]: No. In fact, this is what blocked merging the PR two months ago. Since then we've decided that having some macOS CI on Github actions (even though it lacks some things) is better than not having it at all.

**Review by JesusPoderoso — APPROVED** [review]
LGTM
Please, remember to update all `@feature/fastdds_mac_ci_support` to `@main` before merging

**Review by JesusPoderoso — APPROVED** [review]

**Pull request merged**
